### DISCUSSION TITLE
Constant Function Market Makers

### DISCUSSION THREAD

**Discussion by said017** [Jeiwan/uniswapv3-book#54] [Milestone 0. Introduction]
Hi,

I'm wondering if the statement number 4 in [trade function](https://uniswapv3book.com/docs/introduction/constant-function-market-maker/#the-trade-function) explanation is true "The pool also takes a small fee (r) from the amount of token 0 we gave", should'nt it supposed to be r = 1 - swap fee ?

**Comment by Jeiwan** [comment]
You're correct! Thanks! Fixed: https://github.com/Jeiwan/uniswapv3-book/issues/59
